src/lib.rs

```diff
@@ -325,12 +328,27 @@ mod device_context {
         d.segment_id()
     }
 
-    pub fn private_device_key(d: &DeviceContext) -> PrivateKey {
-        d.private_device_key().clone()
+    pub fn device_private_key(d: &DeviceContext) -> PrivateKey {
+        d.device_private_key().clone()
     }
 
-    pub fn signing_keys(d: &DeviceContext) -> DeviceSigningKeyPair {
-        d.signing_keys().clone()
+    pub fn signing_private_key(d: &DeviceContext) -> DeviceSigningKeyPair {
+        d.signing_private_key().clone()
+    }
+
+    pub fn device_id(d: &DeviceContext) -> DeviceId {
+        d.device_id().clone()
+    }
+
+    pub fn to_json_string(d: &DeviceContext) -> String {
+        //Colt: you should research this unwrap
+        serde_json::to_string(d).unwrap()
```

Comment:
Serialization can fail. Should this just return a Result?

Reply:
@clintfred How does serialization on a struct fail? As far as I can tell it can only fail if a member type can't be serialized or if the keys aren't strings. Neither of those things can happen (we have tests in ironoxide to prove that even more than the types).

I couldn't think of a reason to propagate that error. Is there something else?

Reply:
After giving this more thought, I agree that just doing a `.expect("DeviceContext should always serialize to JSON")` is probably a good way to handle this case.

Reply:
Changed to a `.expect`